classification/keras/code/keras_script.py

```diff
@@ -0,0 +1,482 @@
+
+
+# ## Modelling
+
+# (Neptune) Create a new model and model version
+
+# In[52]:
+
+
+from neptune.new.exceptions import NeptuneModelKeyAlreadyExistsError
+
+project_key = project.get_structure()["sys"]["id"].fetch()
+
+try:
+    model = neptune.init_model(name="keras", key="KER")
+    model.stop()
+    model_version = neptune.init_model_version(model=f"{project_key}-KER", name="keras")
+except NeptuneModelKeyAlreadyExistsError:
+    model_version = neptune.init_model_version(model=f"{project_key}-KER", name="keras")
+
+
+# ### Build a model
+
+# In[53]:
+
+
+model_params = {
+    "dropout": 0.5,
+    "strides": 3,
+    "activation": "relu",
+    "kernel_size": 7,
+    "loss": "binary_crossentropy",
+    "optimizer": "adam",
+    "metrics": ["accuracy"],
+}
+
+
+# In[54]:
+
+
+model_version["params"] = model_params
+
+
+# In[55]:
+
+
+from tensorflow.keras import layers
+
+# A integer input for vocab indices.
+inputs = tf.keras.Input(shape=(None,), dtype="int64")
+
+# Next, we add a layer to map those vocab indices into a space of dimensionality
+# 'embedding_dim'.
+x = layers.Embedding(data_params["max_features"], data_params["embedding_dim"])(inputs)
+x = layers.Dropout(model_params["dropout"])(x)
+
+# Conv1D + global max pooling
+x = layers.Conv1D(
+    data_params["embedding_dim"],
+    model_params["kernel_size"],
+    padding="valid",
+    activation=model_params["activation"],
+    strides=model_params["strides"],
+)(x)
+x = layers.Conv1D(
+    data_params["embedding_dim"],
+    model_params["kernel_size"],
+    padding="valid",
+    activation=model_params["activation"],
+    strides=model_params["strides"],
+)(x)
+x = layers.GlobalMaxPooling1D()(x)
+
+# We add a vanilla hidden layer:
+x = layers.Dense(data_params["embedding_dim"], activation=model_params["activation"])(x)
+x = layers.Dropout(model_params["dropout"])(x)
+
+# We project onto a single unit output layer, and squash it with a sigmoid:
+predictions = layers.Dense(1, activation="sigmoid", name="predictions")(x)
+
+keras_model = tf.keras.Model(inputs, predictions)
+
+# Compile the model with binary crossentropy loss and an adam optimizer.
+keras_model.compile(
+    loss=model_params["loss"], optimizer=model_params["optimizer"], metrics=model_params["metrics"]
+)
+
+
+# ### Train the model
+
+# (Neptune) Initialize the Neptune callback
+
+# In[56]:
+
+
+from neptune.new.integrations.tensorflow_keras import NeptuneCallback
+
+neptune_callback = NeptuneCallback(run=run, log_model_diagram=True, log_on_batch=True)
+
+
+# In[57]:
+
+
+training_params = {
+    "epochs": 3,
+}
+
+
+# In[58]:
+
+
+# Fit the model using the train and test datasets.
+keras_model.fit(
+    train_ds, validation_data=val_ds, epochs=training_params["epochs"], callbacks=neptune_callback
+)
+# Training parameters are logged automatically to Neptune
+
+
+# ### Evaluate the model
+
+# In[59]:
+
+
+_, acc = keras_model.evaluate(test_ds, callbacks=neptune_callback)
```

Comment:
I think it would make sense to move this evaluation below where it is used to compare against the prod model.

Reading down, I was trying to figure where this `acc` came from.

Reply:
Yeah, that'd be better.